src/main/fc/fc_msp.c

```diff
@@ -286,15 +286,19 @@ void initActiveBoxIds(void)
         activeBoxIds[activeBoxIdCount++] = BOXHORIZON;
     }
 
+#ifdef BARO
     if (sensors(SENSOR_BARO)) {
         activeBoxIds[activeBoxIdCount++] = BOXBARO;
     }
+#endif
 
+#ifdef MAG
```

Comment:
Even non-mag (but ACC) targets enabled this before. And headfree mode usable even without compass